### ISSUE TITLE
[LAB] Week 2_Problem 1_송준

### ISSUE THREAD

**Issue by songjoon** [vlab-kaist/NN101_23S#66]
### Problem

Week 2_Problem 1

### Source Code

```python
from base64 import b16decode
from calendar import c
from tkinter import N
from random import random 
from typing import Callable

##                         Problem 1                          ##
##                                                            ##
##            Arbitary x_train, y_train are given.            ##
##   Suppose that x and y have linear correlation, y=wx+b.    ##
##     In function training(), you should return [w, b].      ##
##          In function predict(), you should return          ##
##            list y_test corresponding to x_test.            ##
##                  Made by @jangyoujin0917                   ##
##                                                            ##

# NOTE : Feel free to use torch.optim and tensor.

def training(x_train : list[float], y_train : list[float]) -> list[float]: # DO NOT MODIFY FUNCTION NAME
    # Data normalization code (prevents overflow when calculating MSE, prevents underfitting)
    # Note that you need to convert [w, b] to the original scale before returning value
    # w = w * (y_max - y_min)
    # b = b * (y_max - y_min) + y_min
    y_min = min(y_train)
    y_max = max(y_train)
    normalize = lambda y : (y - y_min)/(y_max - y_min)

    alpha = 0.001
    w = 1
    b = 1
    ### IMPLEMENT FROM HERE
    for i in range(1000):
        w -= alpha * derLossWithW(w,b,x_train, y_train)
        b -= alpha * derLossWithB(w,b,x_train, y_train)
    return [w,b]
def derLossWithW(w,b,x_train, y_train):
    cnt = 0
    for i in range(len(x_train)):
        cnt += (w*x_train[i] + b - y_train[i])*x_train[i]
    return cnt / len(x_train)
def derLossWithB(w,b,x_train, y_train):
    cnt = 0
    for i in range(len(x_train)):
        cnt += (w*x_train[i] + b - y_train[i])
    return cnt / len(x_train)
        

def predict(x_train : list[float], y_train : list[float], x_test : list[float]) -> list[float]: # DO NOT MODIFY FUNCTION NAME
    w,b = training(x_train, y_train)
    result = [0]*len(x_test)
    for i in range(len(x_test)):
        result[i] = w*x_test[i]+b
    return result

if __name__ == "__main__":
    x_train = [0.0, 1.0, 2.0, 3.0, 4.0]
    y_train = [2.0, 4.0, 6.0, 8.0, 10.0] # Note that not all test cases give clear line.
    x_test = [5.0, 10.0, 8.0]
    
    w, b = training(x_train, y_train)
    y_test = predict(x_train, y_train, x_test)

    print(w, b)
    print(y_test)


```

### Description

ㅁㄴㅇ

**Comment by jangyoujin0917** [comment]
Since we don't know about the range of x_train and y_train, derivative of loss can be enormous value to update w and b.
Therefore, in this code, weights of test case 2 diverges, resulting the score nan.
I think you can fix this problem by adjusting the learning rate.

**Comment by Dongyeongkim** [comment]
This issue is closed now because of the lack of progress.
